tests/unit/test_admin.py

```diff
@@ -123,3 +123,45 @@ def test_email_in_change_page(self):
         url = reverse('admin:hastexo_stack_change', args=(self.stack.id,))
         response = self.client.get(url)
         self.assertContains(response, self.email)
+
+    def test_mark_suspended(self):
+        data = {
+            'action': 'mark_suspended', '_selected_action': [self.stack.id, ]}
+        url = reverse('admin:hastexo_stack_changelist')
+        response = self.client.post(url, data, follow=True)
+        self.assertEqual(response.status_code, 200)
+
+        stack = Stack.objects.get(pk=self.stack.id)
+        self.assertEqual(stack.status, "SUSPEND_COMPLETE")
```

Comment:
Might want to check for a non-empty provider here.

Reply:
done.